src/model/generator.py

```diff
@@ -0,0 +1,21 @@
+"""
+Abstract Base Class of a Generator
+"""
+from abc import ABC
+
+
+class Generator(ABC):
+    """
+    Abstract Base Class of a Generator
+
+    This can be used as interface to abstract from different generating models
+    """
+
+    def __int__(self):
+        self.type = "abstract"
+
+    def generate(self):
```

Comment:
```suggestion
    def generate(self, output_file_path):
```

Don't we need an output path here, so its clear where to write the generated file.

Reply:
i dont think we need it, because the user does not need to change it, so it can be static.

Reply:
Maybe let's just add a optional parameter, that allows to set it, but has a default path as well. What do you think?